### PR TITLE
google-app-engine-go-sdk,google-cloud-sdk,google-compute-engine: Refactor to python3

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Working on #101964, focusing here on google tools

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102284` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>google-app-engine-go-sdk</li>
    <li>google-cloud-sdk (google-cloud-sdk-gce)</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4231,14 +4231,16 @@ in
...
+    python = python3;
+  };
   google-cloud-sdk-gce = google-cloud-sdk.override { with-gce = true; };
 
   google-fonts = callPackage ../data/fonts/google-fonts { };
 
   google-clasp = callPackage ../development/misc/google-clasp { };
 
-  google-compute-engine = python2.pkgs.google-compute-engine;
+  google-compute-engine = python3.pkgs.google-compute-engine;
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:4243]:
```suggestion
  google-compute-engine = with python3.pkgs; toPythonApplication google-compute-engine;
```

**Review by jonringer — COMMENTED** [review]
othewise LGTM

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (freezeboy) [reply pkgs/top-level/all-packages.nix:4243]: I'll add this fix

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/102284
3 packages built:
google-app-engine-go-sdk google-cloud-sdk google-compute-engine
```

**Pull request merged**